server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -1309,6 +1330,26 @@ public void failEngine(String reason, @Nullable Exception failure) {
         }
     }
 
+    /**
+     * Cleanup all unreferenced files generated during failed segment merge. This resets shard state to last Lucene
+     * commit.
+     */
+    private void cleanUpUnreferencedFiles() {
+        try (
+            IndexWriter writer = new IndexWriter(
```

Comment:
As per my understanding we don't need any extra changes but did we evaluated this approach when the remote store feature is enabled ?

Reply:
segments are only uploaded after the refresh phase . this is failing much before and shouldn't affect remote store.